test/.env

```diff
@@ -5,6 +5,12 @@ MACHINE=machine_env
 MACHINE_EXPAND=$MACHINE
 UNDEFINED_EXPAND=$UNDEFINED_ENV_KEY
 ESCAPED_EXPAND=\$ESCAPED
+DEFINED_EXPAND_WITH_DEFAULT=${MACHINE:-default}
+DEFINED_EXPAND_WITH_DEFAULT_NESTED=${MACHINE:-${UNDEFINED_ENV_KEY:-default}}
+UNDEFINED_EXPAND_WITH_DEFINED_NESTED=${UNDEFINED_ENV_KEY:-${MACHINE:-default}}
+UNDEFINED_EXPAND_WITH_DEFAULT=${UNDEFINED_ENV_KEY:-default}
+UNDEFINED_EXPAND_WITH_DEFAULT_NESTED=${UNDEFINED_ENV_KEY:-${UNDEFINED_ENV_KEY_2:-default}}
+UNDEFINED_EXPAND_WITH_DEFAULT_NESTED_TWICE=${UNDEFINED_ENV_KEY:-${UNDEFINED_ENV_KEY_2${UNDEFINED_ENV_KEY_3:-default}}}
```

Comment:
@vantreeseba may you help me understand how is this supposed to work? Why are `UNDEFINED_ENV_KEY2` and `${UNDEFINED_ENV_KEY_3:-default}` being concatenated?

Reply:
That actually looks like my test was missing a :-

So it should look like
```bash
UNDEFINED_EXPAND_WITH_DEFAULT_NESTED_TWICE=${UNDEFINED_ENV_KEY:-${UNDEFINED_ENV_KEY_2:-${UNDEFINED_ENV_KEY_3:-default}}}
```

Idea being, it's a n, n+1,n+2 test to do a "proof by induction".

I can fix it, or if you want to fix it in your branch is fine, whichever everyone thinks will be easier/better.